processor/src/main/java/org/mapstruct/ap/processor/CdiComponentProcessor.java

```diff
@@ -44,4 +44,9 @@ protected Annotation getTypeAnnotation() {
     protected Annotation getMapperReferenceAnnotation() {
         return new Annotation( getTypeFactory().getType( "javax.inject.Inject" ) );
     }
+
+    @Override
+    protected boolean shouldDecoratorBeRemoved() {
```

Comment:
Thinking about it, should we raise a warning in case `@DecoratedWith` is used with the "cdi" component model, stating that this isn't necessary and `@DecoratedWith` will be ignored in this case?